### PR TITLE
slack-post-msg action v1.0.3

### PR DESCRIPTION
**Pull request by shoukoo**
# v1.0.3
## Changelogs
- Remove the ability to send custom messages.
- Construct messages in the action
- Include examples in README 
- GitHub context and channel ID are required in the action

## Usage
```yaml


name: first
on:
  push:

jobs:
  test1:
    runs-on: [self-hosted,zendesk-stable]
    strategy:
      matrix:
        task:
          - test1
          - test2
          - test3
    steps:
      - name: ${{ matrix.tasks }}
        run: |
          exit 1

  slack:
    runs-on: [self-hosted, zendesk-stable]
    needs: [test1]
    if: ${{ always() }}
    steps:
      - name: Notify slack
        if: ${{ contains(needs.test1.result, 'failure') }}
        env:
          SLACK_BOT_TOKEN: ${{ secrets.SLACK_TOKEN }}
          GITHUB_CONTEXT: ${{ toJson(github) }}
          STATUS: failure
          CHANNEL_ID: C014024BXX
        uses: zendesk/ga/slack-post-msg@v1
```

## Notifications
image
image

### CODE REVIEW

File: slack-post-msg/entrypoint.sh [hunk 1]
```diff
@@ -1,19 +1,120 @@
...
+run_no=$(echo $GITHUB_CONTEXT | jq --raw-output ".run_number")
+repository=$( echo $GITHUB_CONTEXT | jq --raw-output ".repository")
+run_url="https://github.com/$repository/actions/runs/$run_id"
+ref=$(echo $GITHUB_CONTEXT | jq --raw-output ".ref")
+branch=${ref/refs\/heads\/}
+sha=$(echo $GITHUB_CONTEXT | jq --raw-output ".sha")
+short_sha=${sha:0:8}
+actor=$(echo $GITHUB_CONTEXT | jq --raw-output ".actor")
+workflow=$(echo $GITHUB_CONTEXT | jq --raw-output ".workflow")
+event=$(echo $GITHUB_CONTEXT | jq --raw-output ".event_name")
```
Comment (magnetikonline) [inline slack-post-msg/entrypoint.sh:30]: I'd probably err for `echo "$GITHUB_CONTEXT"` on all these.

File: slack-post-msg/entrypoint.sh [hunk 2]
```diff
@@ -1,19 +1,120 @@
...
+
+
+style="default"
+result="failed"
+icon=":github-approved:"
+
+if [[ $STATUS == "failure" ]]; then
```
Comment (magnetikonline) [inline slack-post-msg/entrypoint.sh:37]: You're not in a bash shell - you're in bourne.

Needs `#!/bin/bash -e` at top of script.

File: slack-post-msg/entrypoint.sh [hunk 3]
```diff
@@ -1,19 +1,120 @@
...
+  style="danger"
+  icon=":github-changes-requested:"
+fi
+
+if [[ $STATUS == "success" ]]; then
```
Comment (magnetikonline) [inline slack-post-msg/entrypoint.sh:42]: Same.

You're not in a bash shell - you're in bourne.

Needs `#!/bin/bash -e` at top of script.

File: slack-post-msg/entrypoint.sh (hunk 1 above)
Comment (shoukoo) [reply slack-post-msg/entrypoint.sh:30]: sorry, what's err?

File: slack-post-msg/entrypoint.sh [hunk 4]
```diff
@@ -1,19 +1,120 @@
...
-set -eu
+
+STATUS=${STATUS:-failure}
 
 if test -z "$SLACK_BOT_TOKEN"; then
   echo "Set the SLACK_BOT_TOKEN secret."
   exit 1
 fi
 
+if test -z "$CHANNEL_ID"; then
```
Comment (magnetikonline) [inline slack-post-msg/entrypoint.sh:10]:
```suggestion
if [[ -z $CHANNEL_ID ]]; then
```

If using Bash, can do this with an internal - no need to call `test`.

File: slack-post-msg/entrypoint.sh [hunk 5]
```diff
@@ -1,19 +1,120 @@
...
+  echo "Set the CHANNEL_ID secret."
+  exit 1
+fi
 
-if test -z "$GITHUB_BODY"; then
-  echo "Set the SLACK_BOT_TOKEN secret."
+if test -z "$GITHUB_CONTEXT"; then
```
Comment (magnetikonline) [inline slack-post-msg/entrypoint.sh:15]:
```suggestion
if [[ -z $GITHUB_CONTEXT ]]; then
```

If using Bash, can do this with an internal - no need to call `test`.

File: slack-post-msg/entrypoint.sh [hunk 6]
```diff
@@ -1,19 +1,120 @@
...
+resp=$(curl -X POST https://slack.com/api/chat.postMessage \
      -H "Content-type: application/json" \
      -H "Authorization: Bearer $SLACK_BOT_TOKEN" \
-     -d "$*" \
-     https://slack.com/api/chat.postMessage
+     -d "$body")
+
+ok=$( echo $resp | jq --raw-output ".ok")
+
+if [[ $ok == "false" ]]; then
```
Comment (magnetikonline) [inline slack-post-msg/entrypoint.sh:118]: Same here - need bash shell.

Also, maybe add an error message?

File: slack-post-msg/entrypoint.sh [hunk 7]
```diff
@@ -1,19 +1,120 @@
...
+  exit 1
```
Comment (magnetikonline) [inline slack-post-msg/entrypoint.sh:119]: Some inconsistent indenting through this script.

File: slack-post-msg/Dockerfile [hunk 8]
```diff
@@ -1,14 +1,15 @@
-FROM curlimages/curl:7.68.0
+FROM alpine
```
Comment (magnetikonline) [inline slack-post-msg/Dockerfile:1]: Do you want to pick a version/label here?

File: slack-post-msg/entrypoint.sh (hunk 1 above)
Comment (magnetikonline) [reply slack-post-msg/entrypoint.sh:30]: hahah - I mean "errrrrr" :D I was "speaking".... 😄 

Yeah I'd quote all those.

**Comment by sfdye** [comment]
I am thinking you can actually use https://github.com/github/super-linter on this repo 🤔

File: slack-post-msg/entrypoint.sh (hunk 1 above)
Comment (shoukoo) [reply slack-post-msg/entrypoint.sh:30]: Sorry @magnetikonline,  I prefer them not quoted

File: slack-post-msg/entrypoint.sh (hunk 2 above)
Comment (shoukoo) [reply slack-post-msg/entrypoint.sh:37]: Updated to #!/bin/sh -e

File: slack-post-msg/entrypoint.sh (hunk 3 above)
Comment (shoukoo) [reply slack-post-msg/entrypoint.sh:42]: Updated to #!/bin/sh -e

**Comment by shoukoo** [comment]
> I am thinking you can actually use https://github.com/github/super-linter on this repo 🤔
Thanks for the suggestion, I might use this in the future.

File: slack-post-msg/entrypoint.sh (hunk 1 above)
Comment (magnetikonline) [reply slack-post-msg/entrypoint.sh:30]: Fair enough - but for reasons like this you should.... @shoukoo 

```sh
echo "this is * a star"
> this is * a star

echo this is * a star
> this is 00archive alias backup.filter backup.sh devupdate.sh gistreset.sh gitfetchall.sh gitgcall.sh gitlistobjectbysize.sh a star
``` 
In the second instance - that asterix is globbing files in the current directory.

Comment (shoukoo) [reply slack-post-msg/entrypoint.sh]: yeah, very fair point. updated

**Pull request merged**